File: Array/running_sum_1d_array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
function runningSum(nums) {
  const result = [];
  let sum = 0;
  for (let i = 0; i < nums.length; i++) {
    sum += nums[i];
    result.push(sum);
  }
  return result;
}

const nums = [1, 1, 1, 1, 1];
const output = runningSum(nums);
console.log(output);

File: Array/running_sum_1d_array.py
from typing import List

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        result = []
        sum = 0
        for i in range(0, len(nums)):
            sum += nums[i]
            result.append(sum)
        return result
    
sol = Solution()
print(sol.runningSum([1, 1, 1, 1, 1]))